prelude/runtime/runtime64_specific.cpp

```diff
@@ -270,18 +270,36 @@ void SKIP_unsetenv(char* name) {
 void SKIP_memory_init(int pargc, char** pargv);
 void sk_persist_consts();
 
-int main(int pargc, char** pargv) {
+void sk_init(int pargc, char** pargv) {
   sk_saved_obstack_t* saved;
   argc = pargc;
   argv = pargv;
-  std::set_terminate(terminate);
   SKIP_memory_init(pargc, pargv);
   saved = SKIP_new_Obstack();
   SKIP_initializeSkip();
   sk_persist_consts();
   SKIP_destroy_Obstack(saved);
+}
+
+#ifdef SKIP_LIBRARY
+__attribute__((constructor)) static void lib_init() {
```

Comment:
There are valid points for not making this a constructor here, and letting downstream choose when it decides to init the skip runtime. Merging as is for now, but this will probably be revisited soon (removing `__attribute__((constructor))`, removing `static`, and renaming the function to something like `SKIP_init()`).